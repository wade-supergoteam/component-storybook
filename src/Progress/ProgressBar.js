// @flow

import styled, { css, keyframes } from "styled-components";
import { gradientStriped } from "../utils/gradients";
import { transition } from "../utils/transition";

import themeProp from "../utils/theme";

import {
  progressHeight,
  progressBarColor,
  progressBarBg,
  progressBarAnimationTiming,
  progressBarTransition,
} from "./default-theme";

const progressBarStripes = keyframes`
  from { background-position: ${progressHeight} 0; }
  to { background-position: 0 0; }
`;

const ProgressBar = styled.div.attrs({
  role: "progressbar",
})`
  height: ${themeProp("progressHeight", progressHeight)};
  width: ${props => props.width};
  line-height: ${themeProp("progressHeight", progressHeight)};
  color: ${themeProp("progressBarColor", progressBarColor)};
  background-color: ${themeProp("progressBarBg", progressBarBg)};
  ${transition(themeProp("progressBarTransition", progressBarTransition))};

  ${props =>
    props.striped &&
    css`
      ${gradientStriped()}
      background-size:
        ${themeProp("progressHeight", progressHeight)}
        ${themeProp("progressHeight", progressHeight)};
  `};

  ${props =>
    props.animated &&
    css`
      animation: ${progressBarStripes} ${themeProp("progressBarAnimationTiming", progressBarAnimationTiming)};
    `};
`;

ProgressBar.defaultProps = {};

export default ProgressBar;
