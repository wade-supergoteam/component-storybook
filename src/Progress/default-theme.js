// @flow

export const progressHeight = "1rem";
export const progressFontSize = ".75rem";
export const progressBg = "#e9ecef";
export const progressBorderRadius = ".25rem";
export const progressBoxShadow = "inset 0 .1rem .1rem rgba(0,0,0,0.1)";
export const progressBarColor = "#fff";
export const progressBarBg = "#007bff";
export const progressBarAnimationTiming = "1s linear infinite";
export const progressBarTransition = "width .6s ease";
