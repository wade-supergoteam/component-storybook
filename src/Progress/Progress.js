// @flow
import React, { cloneElement } from "react";
import styled from "styled-components";
// import PropTypes from 'prop-types';

import { borderRadius } from "../utils/border-radius";
import { boxShadow } from "../utils/box-shadow";

import ProgressBar from "./ProgressBar";

import themeProp from "../utils/theme";

import { progressHeight, progressFontSize, progressBg, progressBorderRadius, progressBoxShadow } from "./default-theme";

const StyledProgress = styled.div`
  display: flex;
  overflow: hidden; // force rounded corners by cropping it
  font-size: ${themeProp("progressFontSize", progressFontSize)};
  line-height: ${themeProp("progressHeight", progressHeight)};
  text-align: center;
  background-color: ${themeProp("progressBg", progressBg)};
  ${borderRadius(themeProp("progressBorderRadius", progressBorderRadius))};
  ${boxShadow(themeProp("progressBoxShadow", progressBoxShadow))};
`;

const Progress = props =>
  props.children ? (
    cloneElement(<StyledProgress />, props)
  ) : (
    <StyledProgress>
      <ProgressBar width={props.width} striped={props.striped} animated={props.animated} />
    </StyledProgress>
  );

Progress.Bar = ProgressBar;

Progress.defaultProps = {};
Progress.propTypes = {};

export default Progress;
