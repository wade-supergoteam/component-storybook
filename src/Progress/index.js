import Progress from "./Progress";
import ProgressBar from "./ProgressBar";

export { ProgressBar };
export default Progress;
