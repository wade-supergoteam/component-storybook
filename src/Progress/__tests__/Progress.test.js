import "jest-styled-components";
import React from "react";
import renderer from "react-test-renderer";
import Progress from "..";

test("Basic Progress", () => {
  const wrapper = renderer.create(<Progress />);
  expect(wrapper).toMatchSnapshot();
});
