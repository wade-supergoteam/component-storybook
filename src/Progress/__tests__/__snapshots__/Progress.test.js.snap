// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Progress 1`] = `
<div
  className="sc-bwzfXH djjGAO"
>
  <div
    className="sc-bdVaJa duOSIu"
    role="progressbar"
    width={undefined}
  />
</div>
`;
